diff --git a/CMakeLists.txt b/CMakeLists.txt
index 58f9aff..894a3ea 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -25,7 +25,7 @@ juce_add_plugin(Hera
   PLUGIN_MANUFACTURER_CODE AcEl
   PRODUCT_NAME "Hera"
   COMPANY_NAME "A/C Electronics"
-  FORMATS AU VST3
+  FORMATS AU Standalone
   IS_SYNTH TRUE
   NEEDS_MIDI_INPUT TRUE
   NEEDS_MIDI_OUTPUT FALSE
@@ -147,3 +147,38 @@ target_link_libraries(Hera
 if(CMAKE_CXX_COMPILER_ID MATCHES "GNU|Clang")
   target_compile_options(Hera PRIVATE "-Wall")
 endif()
+
+# begin JUCEAAP specific -->
+# It is not beautiful, but there does not seem to be any way to add custom include directory for modules.
+include_directories(
+        "${AAP_DIR}/include"
+)
+
+juce_add_module(${AAP_JUCE_DIR}/modules/juceaap_audio_plugin_client)
+target_link_libraries(Hera
+        PRIVATE
+        juceaap_audio_plugin_client
+        )
+
+if (ANDROID)
+
+# dependencies
+find_library(log "log")
+find_library(android "android")
+find_library(glesv2 "GLESv2")
+find_library(egl "EGL")
+set(cpufeatures_lib "cpufeatures")
+set(oboe_lib "oboe")
+set(CMAKE_CXX_FLAGS_RELEASE "-O3")
+
+target_link_libraries(Hera
+        PRIVATE
+        ${log}
+        ${android}
+        ${glesv2}
+        ${egl}
+        ${cpufeatures_lib}
+        )
+
+endif (ANDROID)
+# <-- end JUCEAAP specific
